src/main/java/roomescape/domain/Person.java

```diff
@@ -1,13 +1,27 @@
 package roomescape.domain;
 
 public record Person(String name) {
+
+    private static final int MAX_NAME_LENGTH = 7;
+
     public Person {
-        invalidName(name);
+        validateEmptyName(name);
+        validateNameLength(name);
```

Comment:
```suggestion
        validateEmptyName();
        validateNameLength();
```

인자를 주지 않아도 되는데, 어떻게 생각하시나요?

Reply:
저는 말씀해주신 방식이 this.name에 값을 할당한 이후, 검증을 시도하는 방식이라 생각해 하지 않았습니다. 하지만 알아보니 해당 방식도 검증 이후에 객체가 생성된다는 것을 알았습니다. 후자의 방식이 더 간편하니 리팩터링 하겠습니다.

Reply:
아~ 이거 record 군요!! 그럼 안되겠네요..ㅠ 동작 원리에 대해서는 `Record`의 `Compact Constructor`를 조사해보시길 바래요~

Reply:
아~ 이거 record 군요!! 그럼 안되겠네요..ㅠ 동작 원리에 대해서는 `Record`의 `Compact Constructor`를 조사해보시길 바래요~

Reply:
record에서 컴팩트 생성자의 경우, 초기화 로직은 가장 마지막에 호출해준다고 합니다. 아직 필드에 할당된 상태가 아니어서 검증 로직이 작동하지 않는다고 하네요.